arex-schedule-web-api/src/main/java/com/arextest/schedule/service/PlanProduceService.java

```diff
@@ -257,5 +264,45 @@ private BuildReplayFailReasonEnum validateToResultReason(BuildPlanValidateResult
         }
     }
 
+    public CommonResponse reRunPlan(String planId) {
+        ReplayPlan replayPlan = replayPlanRepository.query(planId);
+        if (replayPlan == null) {
+            return CommonResponse.badResponse("target plan not found");
+        }
+        int rerunStatus = StageUtils.getStageStatus(replayPlan.getReplayPlanStageList(), PlanStageEnum.RE_RUN);
+        if (rerunStatus == StageStatusEnum.PENDING.getCode() || rerunStatus == StageStatusEnum.ONGOING.getCode()) {
+            return CommonResponse.badResponse("This plan is ReRunning");
+        }
+        replayPlan.setReRun(Boolean.TRUE);
+
+        ConfigurationService.ScheduleConfiguration schedule = configurationService.schedule(replayPlan.getAppId());
+        if (schedule != null) {
+            replayPlan.setReplaySendMaxQps(schedule.getSendMaxQps());
+        }
+
+        planExecutionMonitorImpl.register(replayPlan);
+        try {
+            progressEvent.onReplayPlanReRun(replayPlan);
+
+            progressEvent.onReplayPlanStageUpdate(replayPlan, PlanStageEnum.LOADING_CASE, StageStatusEnum.ONGOING,
+                System.currentTimeMillis(), null, null);
+            boolean reLoad = planConsumePrepareService.prepareAndUpdateFailedActionAndCase(replayPlan);
+            planConsumePrepareService.doResumeOperationDescriptor(replayPlan);
+            progressEvent.onReplayPlanStageUpdate(replayPlan, PlanStageEnum.LOADING_CASE, StageStatusEnum.success(reLoad),
+                null, System.currentTimeMillis(), null);
+            if (!reLoad) {
+                throw new RuntimeException("No case to rerun");
```

Comment:
In the case of no failure case, need it throw an exception directly?

Reply:
In order to enter catch exception logic code.

Reply:
fixed